ngraph_bridge/ngraph_pipelined_tensors.cc

```diff
@@ -0,0 +1,141 @@
+/*******************************************************************************
+ * Copyright 2019 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+
+#include "ngraph_bridge/ngraph_pipelined_tensors.h"
+
+using namespace std;
+namespace ng = ngraph;
+
+namespace tensorflow {
+
+namespace ngraph_bridge {
+
+IndexLibrary::IndexLibrary(size_t depth) : m_depth(depth) {
+  for (size_t i = 0; i < depth; i++) {
+    m_free_depth_indexes.insert(i);
+  }
+}
+
+void IndexLibrary::return_index(size_t id) {
+  if (m_depth == 0) {
+    throw std::runtime_error(
+        "Depth=0, so no one should be calling return_index");
+  }
+  if (id > m_depth - 1) {
+    throw std::runtime_error("Depth = " + to_string(m_depth) +
+                             " but passed an index to return ( = " +
+                             to_string(id) + "), which is too large");
+  }
+  if (is_free(id)) {
+    throw std::runtime_error(
+        "Attempted to return index " + to_string(id) +
+        " but it is already present in the free indices set");
+  }
+
+  insert_to_free_set(id);
+}
+
+int IndexLibrary::get_index() {
+  if (m_depth == 0) {
+    return -1;
+  } else {
+    std::lock_guard<std::mutex> lock(m_mtx);
+    if (m_free_depth_indexes.size() == 0) {
+      return -1;
+    } else {
```

Comment:
optional : dont need else-es where you are returning. It reduces the number of conditions and braces